lib/open_food_network/feature_toggle.rb

```diff
@@ -25,32 +25,50 @@ module OpenFoodNetwork
   #   - if feature? :new_shiny_feature, spree_current_user
   #     = render "new_shiny_feature"
   #
-  class FeatureToggle
+  module FeatureToggle
     def self.enabled?(feature_name, user = nil)
-      new.enabled?(feature_name, user)
+      features = Thread.current[:features] || {}
+
+      if Flipper[feature_name].exist?
+        Flipper.enabled?(feature_name, user)
+      else
+        feature = features.fetch(feature_name, DefaultFeature.new(feature_name))
+        feature.enabled?(user)
+      end
     end
 
     def self.enable(feature_name, &block)
       Thread.current[:features] ||= {}
       Thread.current[:features][feature_name] = Feature.new(block)
     end
+  end
 
-    def initialize
-      @features = Thread.current[:features] || {}
+  class Feature
+    def initialize(block)
+      @block = block
     end
 
-    def enabled?(feature_name, user)
-      if user.present?
-        feature = features.fetch(feature_name, NullFeature.new)
-        feature.enabled?(user)
-      else
-        true?(env_variable_value(feature_name))
-      end
+    def enabled?(user)
+      block.call(user)
     end
 
     private
 
-    attr_reader :features
+    attr_reader :block
+  end
+
+  class DefaultFeature
+    attr_reader :feature_name
+
+    def initialize(feature_name)
+      @feature_name = feature_name
+    end
+
+    def enabled?(_user)
+      true?(env_variable_value(feature_name))
+    end
+
+    private
```

Comment:
I'm not sure I fully understand the need for the `DefaultFeature` class here - is that something required by Flipper? Or leftover from our own implementation?

Reply:
This one is from @mkllnk and yes, comes from our own implementation. I let Maikel answer on this one too.

Reply:
I see this class is implementing the pre-existing ENV-based logic used to have per-instance custom features. IMO this is fine now that we're still using the balances toggle but once that's done, I'd decouple these ENV ones from Flipper completely. No need to abstract things with our `OpenFoodNetwork::FeatureToggle` module.

Reply:
That's our own implementation. I did that to call Flipper first and if Flipper doesn't know about the feature then we revert to our old implementation. I wanted to keep the backwards compatibility here while we introduce Flipper. But I agree with Pau that we will phase out the usage and then we can remove code.

I was also demonstrating how we could set defaults through our own feature toggle implementation because we couldn't set defaults through Flipper in the initializer (we can't write to the database before we migrated the database). But in a subsequent discussion, @jibees and I agree that we won't activate any feature by default. So we won't need that in the future.